js/src/ui/Form/AddressSelect/addressSelect.js

```diff
@@ -15,262 +15,799 @@
+
+    return this.handleDOMAction('inputAddress', 'focus');
+  }
+
+  /**
+   * Filter the given values based on the given
+   * filter
+   */
+  filterValues = (values = [], _filter = '') => {
+    const filter = _filter.toLowerCase();
+
+    return values
+      // Remove empty accounts
+      .filter((a) => a)
+      .filter((account) => {
+        const address = account.address.toLowerCase();
+        const inAddress = address.includes(filter);
+
+        if (!account.name || inAddress) {
+          return inAddress;
+        }
+
+        const name = account.name.toLowerCase();
+        const inName = name.includes(filter);
+        const { meta = {} } = account;
+
+        if (!meta.tags || inName) {
+          return inName;
+        }
+
+        const tags = (meta.tags || []).join('');
+        return tags.includes(filter);
+      })
+      .sort((accA, accB) => {
+        const nameA = accA.name || accA.address;
+        const nameB = accB.name || accB.address;
+
+        return nameA.localeCompare(nameB);
+      });
+  }
+
+  handleInputBlur = () => {
+    this.setState({ inputFocused: false });
+  }
+
+  handleInputFocus = () => {
+    this.setState({ focusedItem: null, inputFocused: true });
+  }
+
+  handleChange = (event = { target: {} }) => {
+    const { value = '' } = event.target;
+    let index = 0;
+
+    window.setTimeout(() => { this.handleTagsOpacity(); });
+
+    const values = this.values
+      .map((category) => {
+        const filteredValues = this
+          .filterValues(category.values, value)
+          .map((value) => {
+            index++;
+            return { ...value, index: parseInt(index) };
+          });
+
+        return {
+          label: category.label,
+          values: filteredValues
+        };
+      })
+      .filter((category) => category.values.length > 0);
+
+    this.setState({
+      values,
+      focusedItem: null,
+      inputValue: value
+    });
+  }
+
+  handleTagsOpacity = () => {
+    Object.values(this.nameRefs).forEach((data) => {
+      if (data.tagsRef.length === 0) {
+        return;
+      }
+
+      const nameEl = ReactDOM.findDOMNode(data.ref);
+
+      if (!nameEl) {
+        return;
+      }
+
+      const nameBounds = nameEl.getBoundingClientRect();
+
+      data.tagsRef.forEach((tagRef) => {
+        const tagEl = ReactDOM.findDOMNode(tagRef);
+
+        if (!tagEl) {
+          return;
+        }
+
+        const tagBounds = tagEl.getBoundingClientRect();
+
+        // Hide if haven't at least a 10px margin
+        tagEl.style.opacity = (tagBounds.left > nameBounds.right + 10)
+          ? 1
+          : 0;
+      });
+    });
+  }
+
+  preventEvent = (e) => {
+    e.preventDefault();
+    e.stopPropagation();
+  }
+}
+
+function mapStateToProps (state) {
+  const { accountsInfo } = state.personal;
+  const { balances } = state.balances;
+
+  return {
+    accountsInfo,
+    balances
   };
 }
+
+export default connect(
+  mapStateToProps
+)(AddressSelect);
+
```

Comment:
At this size the component has reached the point of being a beast and not being maintainable at all. In addition, none of the sub-elements can be re-used, i.e. account cards are something I could use in my PRs (once going down the same route), however cannot extract as it is.